pylint/config/config_initialization.py

```diff
@@ -141,7 +145,8 @@ def _config_initialization(
     linter._parse_error_mode()
 
     # Link the base Namespace object on the current directory
-    linter._directory_namespaces[Path(".").resolve()] = (linter.config, {})
+    if Path(".").resolve() not in linter._directory_namespaces:
+        linter._directory_namespaces[Path(".").resolve()] = (linter.config, {})
```

Comment:
This should not have the `if` I think? `_config_initialization` should be called once? Even for multi-dir?

Reply:
`_config_initialization` has some non-trivial logic for parsing several possible variants of configs into namespace, merging it with command-line arguments, configuring plugins and reporting errors during this process. So it was convenient to reuse all this logic for parsing additional configs, and `_config_initialization` is going to be called for each new config in subsequent changes

Reply:
Hmm but shouldn't we pass the path of the current config file to this function then? On its own this if statement doesn't make a lot of sense.

Reply:
Exactly - the path of the current config file is passed to `_config_initialization` in `Run.__init__`, and paths of new config files will be passed there in `register_local_config`.

The idea of this `if` is that first time when `_config_initialization` is called, it parses config from working directory, and this config should be saved to `linter._directory_namespaces` to avoid additional processing of special cases. But next times when `_config_initialization` is called, it shouldn't overwrite config for working directory with values from new files.